src/Android/Accessibility/AccessibilityHelpers.cs

```diff
@@ -280,17 +291,50 @@ public static void FillEditText(AccessibilityNodeInfo editTextNode, string value
             return nodes;
         }
 
-        public static void GetNodesAndFill(AccessibilityNodeInfo root, AccessibilityEvent e,
-            IEnumerable<AccessibilityNodeInfo> passwordNodes)
+        public static AccessibilityNodeInfo GetUsernameEditText(string uriString, 
+            IEnumerable<AccessibilityNodeInfo> allEditTexts)
         {
-            var allEditTexts = GetWindowNodes(root, e, n => EditText(n), false);
-            var usernameEditText = GetUsernameEditTextIfPasswordExists(allEditTexts);
-            FillCredentials(usernameEditText, passwordNodes);
-            allEditTexts.Dispose();
-            usernameEditText = null;
-        }
+            string uriKey = null;
+            string uriLocalPath = null;
+            if (Uri.TryCreate(uriString, UriKind.Absolute, out var uri))
+            {
+                uriKey = uri.Authority;
+                uriLocalPath = uri.LocalPath;
+            }
 
-        public static AccessibilityNodeInfo GetUsernameEditTextIfPasswordExists(
+            if (!string.IsNullOrEmpty(uriKey))
+            {
+                // Uncomment this to log values necessary for username field discovery
+                // foreach (var editText in allEditTexts)
+                // {
+                //     System.Diagnostics.Debug.WriteLine(">>> uriKey: {0}, uriLocalPath: {1}, viewId: {2}", uriKey,
+                //         uriLocalPath, editText.ViewIdResourceName);
+                // }
+
+                if (KnownUsernameFields.ContainsKey(uriKey))
+                {
+                    var usernameField = KnownUsernameFields[uriKey];
+                    if (uriLocalPath.EndsWith(usernameField.UriPathEnd))
+                    {
+                        foreach (var editText in allEditTexts)
+                        {
+                            foreach (var usernameViewId in usernameField.UsernameViewId.Split(","))
+                            {
+                                if (usernameViewId == editText.ViewIdResourceName)
```

Comment:
Should this be a case-insensitive comparison?

Reply:
Hm, not sure.  My intent was for an exact match, but I'm curious what your thoughts are.

Reply:
Actually, I think exact match makes sense, HTML is case-sensitive for Ids so we probably should be as well, thanks.